src/pybind/mgr/dashboard/frontend/package.json

```diff
@@ -76,8 +76,11 @@
     "ngx-toastr": "17.0.2",
     "rxjs": "6.6.3",
     "simplebar-angular": "2.3.6",
+    "stream": "0.0.2",
```

Comment:
I cant see where its used

Reply:
Not being used, forgot to remove them, thid and the others, thanks for pointing it out!

Reply:
Ack..I am not sure if this is something still maintained as last commit is 8 yrs ago
PTAL https://github.com/juliangruber/stream/

Reply:
This dependencie comes from the use of sax.js which comes from the xml2js, so not sure how critical that is...

I have to say, if I get rid of it, it works, but get the following warning:
```
./node_modules/sax/lib/sax.js:139:13-37 - Warning: Module not found: Error: Can't resolve 'stream' in '/home/pegonzal/ceph/ceph/src/pybind/mgr/dashboard/frontend/node_modules/sax/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

```

So what do you think? I honestly would just keep it

Reply:
Okay, may we can look for native libraries and if its complex in js then perhaps move the parsing to backend.

Reply:
When I looked into it, the easiest approach I saw was this one. That's why I went with it.
I personally think it's easier to keep the dependency in the frontend and parse the xml from the frontend directly.
And about native libraries... do you know any to achive this?

Reply:
Here is the native example that I found on MDN:
DOMparser I see from JS. Here is the MDN doc: https://developer.mozilla.org/en-US/docs/Web/XML/Parsing_and_serializing_XML#parsing_strings_into_dom_trees

I just hesitant on this due to the third-part library adding some deps which are no longer maintained. It can come as a toll later using such libs. :)

Reply:
yeah, using an 8 year old unmaintained depedency is a bit troubling especially if it poses any security issues. I couldn't find anything about it yet. But I also saw something about using [timers-browserify](https://www.npmjs.com/package/timers-browserify) and [stream-browserify](https://www.npmjs.com/package/stream-browserify) instead of the current ones to fix the issue in angular. Have you tried that?
https://stackoverflow.com/a/55788098

Reply:
Thank you both for the input, for now I'll make use of the [timers-browserify](https://www.npmjs.com/package/timers-browserify) and [stream-browserify](https://www.npmjs.com/package/stream-browserify) as suggested by @nizamial09, and since some work may be done in the near future to accomodate the ACLs better as stated in the tracker https://tracker.ceph.com/issues/64297, I've also notted down your input @afreen23 to look into removing the xml2js dependency for a native solution.